src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -962,7 +961,20 @@ object FunctionRegistry {
   registerUnaryNAFilteredCollectionMethod("mean", { (x: TraversableOnce[Float]) => if (x.nonEmpty) box(x.sum / x.size.toDouble) else null }, meanDocstring)(float32Hr, boxedFloat64Hr)
   registerUnaryNAFilteredCollectionMethod("mean", { (x: TraversableOnce[Double]) => if (x.nonEmpty) box(x.sum / x.size.toDouble) else null }, meanDocstring)(float64Hr, boxedFloat64Hr)
 
-  register("range", { (x: Int) => 0 until x: IndexedSeq[Int] },
+  register("range", { (x: Int) =>
```

Comment:
This is probably going to slow down real pipelines by a significant amount. I actually think when we push expr fully down into RVs, this is going to be something that gets worse unless we explicitly introduce range optimizations.

Reply:
For context, we had a 24-hour sprint to fix Hail's abysmal performance on the gnomAD v1 analysis pipeline, and something like 95% of the time was spent in the range implementation that was basically `(0 until N).toArray: IndexedSeq[Int]`

Reply:
here's the commit: https://github.com/hail-is/hail/commit/414d2af17024db469e44774ad194d25fe998a070

Reply:
Ha!  As far as I can tell, this significantly sped up the gnomAD pipeline that was spending a ton of time in annotate_alleles downcoding PLs with:

```
  range(3).map(i => range(g.PL.length).filter(j => downcode(Call(j), aIndex) == Call(i)).map(j => g.PL[j]).min())
```

I can try re-benchmark if you'd like.

In @danking's new expr stuff chained Array, etc., operations won't actually generate any intermediates so this will be moot soon.

Reply:
Maybe the fact it's an Array instead of a Range => Vector is actually improving it for this pipeline. I can dig through Slack history to see what Konrad was doing last year where this made a huge difference

Reply:
There's a fair bit of indirection in `(0 to n).toArray`, including allocation of an iterator.

Since arrays are immutable in Hail, do you think we'd see some benefit to interning all the ranges from 0 to 10?

Reply:
absolutely. Probably could do a bit more, even.

Reply:
that's a very good idea, I bet that would speed things up a bunch right now

Reply:
I'm going to leave leave additional benchmarking improvements and @danking's comment for another PR once @patrick-schultz gives the OK.